zproject/backends.py

```diff
@@ -143,7 +144,14 @@ def get_or_create_user(self, username, ldap_user):
         try:
             return get_user_profile_by_email(username), False
         except UserProfile.DoesNotExist:
-            return UserProfile(), False
+            realm = Realm.objects.get(domain=settings.LDAP_APPEND_DOMAIN)
```

Comment:
If you look at the other functions in the file, I think we support LDAP_APPEND_DOMAIN=None (in which case the LDAP usernames are email addresses).  Also this is written in a way where it doesn't support realm aliases.  So I think you'll want to use 

resolve_email_to_domain(ldap_to_django_username(ldap_user)) 

here instead to get the realm.